drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json

```diff
@@ -269,34 +269,108 @@
     {
       "type": "GROUP",
       "link_id": "income-source",
-      "text": "Income and Source",
+      "text": "Income and Sources",
       "data_collected_about": "HOH_AND_ADULTS",
       "item": [
         {
-          "type": "CHOICE",
-          "link_id": "4.02.2",
-          "text": "Income from Any Source",
-          "pick_list_reference": "YesNoMissing"
-        },
-        {
-          "type": "BOOLEAN",
-          "link_id": "4.02.3",
-          "text": "Earned Income (i.e., employment income)"
-        },
-        {
-          "type": "CURRENCY",
-          "link_id": "4.02.A",
-          "text": "Monthly Amount",
-          "enable_behavior": "ALL",
-          "bounds": [{
-            "type": "MIN",
-            "value_number": 0
-          }],
-          "enable_when": [
+          "type": "GROUP",
+          "link_id": "income-sources-group",
+          "component": "INPUT_GROUP",
+          "item": [
             {
-              "question": "4.02.3",
-              "operator": "EQUAL",
-              "answer_boolean": true
+              "type": "CHOICE",
+              "link_id": "4.02.2",
+              "text": "Income from Any Source",
+              "pick_list_reference": "YesNoMissingReason"
+            },
+            {
+              "type": "CURRENCY",
+              "link_id": "4.02.A",
+              "text": "Earned Income (i.e., employment income)"
+            },
+            {
+              "type": "CURRENCY",
+              "link_id": "4.02.B",
+              "text": "Unemployment Insurance"
+            },
+            {
+              "type": "CURRENCY",
+              "link_id": "4.02.C",
+              "text": "Supplemental Security Income (SSI)"
+            },
```

Comment:
@eanders @awisema the form only collects 4.02.A-P and not 4.02.3-18 ([reference](https://www.hudexchange.info/programs/hmis/hmis-data-standards/standards/Program_Specific_Data_Elements.htm)). The idea is that the boolean fields can be inferred -- and the spec explicitly says that's OK. I'm not sure about whether that inference logic should live here in the form definition, or whether it should live in the backend processing code.

If we do it at the post-processing step, each of the boolean fields would need to be inferred (based on if the corresponding number is > 0), and the total monthly income would need to be calculated (4.04.18).

If we put the logic in the form, we add some fields that allow us to express "this field value is true if this other field is >0" (similar to existing enableWhen logic) and "this field is the sum of all these other fields."
I think I prefer backend because it's unnecessary to run that logic in the browser, since it doesn't benefit the user in any way.

Reply:
I agree, we should put this in the back-end and require as little hoop jumping as possible on the front-end.  That said, it might be worth having a sum of these other fields read-only display.

Reply:
The frontend component calculates and displays the read-only total amount for this group type (if all the children items are currencies). I considered specifying that read-only sum as an item in the form definition, but decided to leave it in the frontend code for now until/unless I see a pattern emerge.